### PR TITLE
Allows code to compile without ExodusII

### LINKED ISSUE #125: Allow code to compile without Exodus-II
add `#if defined (PETSC_HAVE_EXODUSII)` around the exodus code.

### PR DESCRIPTION
**Pull request by ghammond86**
Added preprocessing to avoid ExodusII code when PETSc is not configured with it.
added condition in regression test makefile to skip ExodusII tests

Fixes #125

### CODE REVIEW

**Comment by bishtgautam** [comment]
@satkarra Can you review this PR to make sure that you can now compile the code with ExodusII?

File: src/tdyio.c [hunk 1]
```diff
@@ -69,6 +71,7 @@ PetscErrorCode TdyIOInitializeExodus(char *ofilename, char *zonalVarNames[], DM
   PetscErrorCode ierr;
```
Comment (bishtgautam) [inline src/tdyio.c:71]: Let's move all local variable definitions inside `ifdef` to avoid warnings

File: src/tdyio.c [hunk 2]
```diff
@@ -90,6 +96,7 @@ PetscErrorCode TdyIOAddExodusTime(char *ofilename, PetscReal time, TDyIO io){
...
```
Comment (bishtgautam) [inline src/tdyio.c:96]: Let's move all local variable definitions inside `ifdef` to avoid warnings

File: src/tdyio.c [hunk 3]
```diff
@@ -108,13 +116,15 @@ PetscErrorCode TdyIOWriteExodusVar(char *ofilename, Vec U, TDyIO io){
   float version;
```
Comment (bishtgautam) [inline src/tdyio.c:116]: Let's move all local variable definitions inside `ifdef` to avoid warnings

**Review by bishtgautam — CHANGES_REQUESTED** [review]
Minor changes requested to avoid warnings

**Review by satkarra — APPROVED** [review]
Works fine on my mac without exodus.

File: src/tdyio.c (hunk 1 above)
Comment (ghammond86) [reply src/tdyio.c:72]: Done (in all locations)

**Review by bishtgautam — APPROVED** [review]

**Pull request merged**